family-dashboard/api/src/main/java/org/phenotips/familydashboard/internal/OmimToHpoMapper.java

```diff
@@ -0,0 +1,248 @@
+     * @return
+     */
+    private boolean foundPatientPhenotypesToSymptomParentsMatch(String symptom)
+    {
+        VocabularyTerm symptomTerm = this.hpoService.getTerm(symptom);
+        if (symptomTerm != null) {
+            for (VocabularyTerm term : symptomTerm.getParents()) {
+                System.out.println(term.getId() + ": " + term.getName());
+                if (this.hpoTerms.contains(term.getId())) {
+                    setCurrentMatch(term);
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks for a match between the actual_symptom and the parent terms of each item in the patient's phenotype set.
+     *
+     * @param symptom the actual symptom that is associated with an OMIM disorder
+     * @return
+     */
+    private boolean foundSymptomToPatientPhenotypeParentsMatch(String symptom)
+    {
+        for (String feature : this.hpoTerms) {
+            VocabularyTerm patientTerm = this.hpoService.getTerm(feature);
+            if (patientTerm != null) {
+
+                for (VocabularyTerm parent : patientTerm.getParents()) {
+                    System.out.println(parent.getId() + ": " + parent.getName());
+                    if (symptom.equals(parent.getId())) {
+                        setCurrentMatch(patientTerm);
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private void setOmimToHpoMapping()
+    {
+        this.omimToHpoMapping = new HashMap<>();
+        for (String term : this.omimTerms) {
+            JSONObject omimResult = getOmimTermResults(term);
+            if (omimResult != null) {
+                JSONArray actualSymptoms = omimResult.getJSONArray("actual_symptom");
+                int threshold = actualSymptoms.length() <= this.matchThreshold ?
+                    actualSymptoms.length() : this.matchThreshold;
+                Map<String, JSONObject> matchedFeatures = new HashMap<>();
+                for (int i = 0; i < this.matchThreshold; i++) {
+                    String symptom = actualSymptoms.getString(i);
+                    if (hasRelevantPhenotype(symptom) && !matchedFeatures.containsKey(getCurrentMatch().getString("id"))) {
+                        matchedFeatures.put(getCurrentMatch().getString("id"), getCurrentMatch());
+                    }
+                }
+                omimToHpoMapping.put(term, matchedFeatures);
+            }
+        }
+    }
+
+    private void setDisorders(JSONObject data)
+    {
+        this.omimTerms = new ArrayList<>();
+        JSONArray array = data.optJSONArray("disorders");
+        if (array != null) {
+            for (Object disorder : array) {
+                if (disorder instanceof JSONObject) {
+                    JSONObject disorderObj = (JSONObject) disorder;
+                    omimTerms.add(disorderObj.optString("id"));
```

Comment:
`optString` means that it may return `null`. Perhaps you want to get the value and check if it's not null before adding it?